packages/frontapp/src/app/components/CategorySearch/index.tsx

```diff
@@ -65,6 +68,143 @@ type itemsType = {
+      ...acc,
+      [material?.author as string]: acc[material?.author as string]
+        ? acc[material?.author as string] + 1
+        : 1,
+    }),
+    {}
+  );
+  const allTypes = availableMaterial?.reduce(
+    (acc: { [type: string]: number }, material: Material) => ({
+      ...acc,
+      [material?.type as string]: acc[material?.type as string]
+        ? acc[material?.type as string] + 1
+        : 1,
+    }),
+    {}
+  );
+  const allAvailability = availableMaterial?.reduce(
+    (acc: { [type: string]: number }, material: Material) => {
+      const lastStatus = material.statuses.slice(-1)[0];
+      const currentStatus = getStatus(lastStatus?.status, material?.created_at);
+      return {
+        ...acc,
+        [currentStatus as string]: acc[currentStatus as string]
+          ? acc[currentStatus as string] + 1
+          : 1,
+      };
+    },
+    {}
+  );
+  useEffect(() => {
+    if (allAuthors && allCategories && allTypes) {
+      setAllFilters([
+        {
+          title: 'Availability',
+          inputType: 'checkbox',
+          id: 1,
+          subAttributes: Object.keys(allAvailability).map((key) => {
+            return {
+              title: key,
+              numberOfItems: allAvailability[key],
+              id: idOfItems++,
+              checked: false,
+            };
+          }),
+        },
+        {
+          title: 'Items',
+          inputType: 'radio',
+          id: 2,
+          subAttributes: Object.keys(allTypes).map((key) => {
+            return {
+              title: key,
+              numberOfItems: allTypes[key],
+              id: idOfItems++,
+              checked: false,
+            };
+          }),
+        },
+        {
+          title: 'Categories',
+          inputType: 'checkbox',
+
+          id: 3,
+          subAttributes: Object.keys(allCategories).map((key) => {
+            return {
+              title: key,
+              numberOfItems: allCategories[key],
+              id: idOfItems++,
+              checked: false,
+            };
+          }),
+        },
+        {
+          title: 'Authors',
+          inputType: 'checkbox',
+
+          id: 4,
+          subAttributes: Object.keys(allAuthors).map((key) => {
+            return {
+              title: key,
+              numberOfItems: allAuthors[key],
+              id: idOfItems++,
+              checked: false,
+            };
+          }),
+        },
+        {
+          title: 'SortBy',
+          inputType: 'radio',
+          id: 5,
+          subAttributes: [
+            {
+              title: 'By date added',
+              numberOfItems: undefined,
+              id: 1,
+              checked: false,
+            },
+            {
+              title: 'By date of writing',
+              numberOfItems: undefined,
+              id: 2,
+              checked: false,
+            },
+          ],
+        },
+      ]);
+    }
```

Comment:
this can be taken out in a separate variable, I think, or come up with something else, because it looks very scary. We need to strive to ensure that useEffect is cleaner. This code is unreadable :)